From 7bf1157e2da02d6e82a693366ccdbdaae8651ff0 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 30 Apr 2015 11:17:19 +0200
Subject: [PATCH 089/118] inspection: recognize xfs as Linux filesystem
 (RHBZ#1216305)

Handle also xfs as file system for Linux systems (like by default
in RHEL 7+) in the old inspection code.
---
 perl/lib/Sys/Guestfs/Lib.pm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/perl/lib/Sys/Guestfs/Lib.pm b/perl/lib/Sys/Guestfs/Lib.pm
index cd03112..ce46b4f 100644
--- a/perl/lib/Sys/Guestfs/Lib.pm
+++ b/perl/lib/Sys/Guestfs/Lib.pm
@@ -434,6 +434,9 @@ sub inspect_partition
     } elsif ($file =~ /ext4 filesystem data/) {
         $r{fstype} = "ext4";
         $r{fsos} = "linux";
+    } elsif ($file =~ /XFS filesystem data/) {
+        $r{fstype} = "xfs";
+        $r{fsos} = "linux";
     } elsif ($file =~ m{Linux/i386 swap file}) {
         $r{fstype} = "swap";
         $r{fsos} = "linux";
-- 
2.7.4

